src/plugins/vis_type_vega/public/expressions/line_vega_spec_fn.ts

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { cloneDeep } from 'lodash';
+import { i18n } from '@osd/i18n';
+import {
+  ExpressionFunctionDefinition,
+  OpenSearchDashboardsDatatable,
+  OpenSearchDashboardsDatatableColumn,
+} from '../../../expressions/public';
+import { VegaVisualizationDependencies } from '../plugin';
+import { VislibDimensions, VisParams } from '../../../visualizations/public';
+
+type Input = OpenSearchDashboardsDatatable;
+type Output = Promise<string>;
+
+interface Arguments {
+  visLayers: string | null;
+  visParams: string;
+  dimensions: string;
+}
+
+export type VegaSpecExpressionFunctionDefinition = ExpressionFunctionDefinition<
+  'line_vega_spec',
+  Input,
+  Arguments,
+  Output
+>;
+
+// Get the first xaxis field as only 1 setup of X Axis will be supported and
+// there won't be support for split series and split chart
+const getXAxisId = (dimensions: any, columns: OpenSearchDashboardsDatatableColumn[]): string => {
+  return columns.filter((column) => column.name === dimensions.x.label)[0].id;
+};
+
+const cleanString = (rawString: string): string => {
+  return rawString.replaceAll('"', '');
+};
+
+const formatDataTable = (
+  datatable: OpenSearchDashboardsDatatable
+): OpenSearchDashboardsDatatable => {
+  datatable.columns.forEach((column) => {
+    // clean quotation marks from names in columns
+    column.name = cleanString(column.name);
+  });
+  return datatable;
+};
+
+const createSpecFromDatatable = (
+  datatable: OpenSearchDashboardsDatatable,
+  visParams: VisParams,
+  dimensions: VislibDimensions
+): object => {
+  // TODO: we can try to use VegaSpec type but it is currently very outdated, where many
+  // of the fields and sub-fields don't have other optional params that we want for customizing.
+  // For now, we make this more loosely-typed by just specifying it as a generic object.
+  const spec = {} as any;
+
+  const legendPosition = visParams.legendPosition;
+
+  spec.$schema = 'https://vega.github.io/schema/vega-lite/v5.json';
+  spec.data = {
+    values: datatable.rows,
+  };
+  spec.config = {
+    view: {
+      stroke: null,
+    },
+    concat: {
+      spacing: 0,
+    },
+    // the circle timeline representing annotations
+    circle: {
+      color: 'blue',
+    },
+    // the vertical line when user hovers over an annotation circle
+    rule: {
+      color: 'red',
+    },
```

Comment:
I think we can remove these from the config for now - i'll add back what's needed in the follow up PR that augments the spec. For example, circle may change to triangle.

Reply:
I almost wonder if it makes sense for the config to come from the vis augmenter? It doesn't seem like any of these are currently coming from the visParams?

Reply:
Yeah good callout. I'll add a note in #3145 to explore that.

I think for the scope of this PR we can keep it here. If it makes sense or is feasible to refactor some of this in `vis_augmenter` I'll follow up on that.

Reply:
Update: added note https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3145#issuecomment-1433462459

Reply:
That is a good point, when we have the PR to augment, we can take a look at that.